modules/graph/fragment/property_graph_types.h

```diff
@@ -546,6 +841,59 @@ class AdjList {
  private:
   const NbrUnit<VID_T, EID_T>* begin_;
   const NbrUnit<VID_T, EID_T>* end_;
+
+  const uint8_t* v_begin_ptr_;
+  const uint8_t* v_end_ptr_;
+  const uint8_t* e_begin_ptr_;
+  const uint8_t* e_end_ptr_;
+  size_t e_size_ = 0;
+  size_t v_size_ = 0;
+  bool encoded_ = false;
+
+  const void** edata_arrays_;
+};
+
+template <typename VID_T, typename EID_T>
+class EncodedAdjList {
+ public:
+  EncodedAdjList()
+      : begin_ptr_(nullptr),
+        end_ptr_(nullptr),
+        size_(0),
+        edata_arrays_(nullptr) {}
+  EncodedAdjList(const uint8_t* ptr, const size_t begin_offset,
+                 const size_t end_offset, const void** edata_arrays) {
+    begin_ptr_ = ptr + begin_offset;
+    end_ptr_ = ptr + end_offset;
+    size_ = end_offset - begin_offset;
+  }
+
+  inline EncodedNbr<VID_T, EID_T> begin() const {
+    return EncodedNbr<VID_T, EID_T>(begin_ptr_, size_, edata_arrays_);
+  }
+
+  inline EncodedNbr<VID_T, EID_T> end() const {
+    return EncodedNbr<VID_T, EID_T>(end_ptr_, 0, edata_arrays_);
+  }
+
+  inline size_t Size() const { return size_; }
+
+  inline bool Empty() const { return size_ == 0; }
+
+  inline bool NotEmpty() const { return size_ != 0; }
+
+  // may be there are bugs.
+  size_t size() const { return size_; }
+
+  // inline const NbrUnit<VID_T, EID_T>* begin_unit() const { return begin_; }
+
+  // inline const NbrUnit<VID_T, EID_T>* end_unit() const { return end_; }
+
+ private:
+  const uint8_t* begin_ptr_;
+  const uint8_t* end_ptr_;
+  size_t size_ = 0;
+
```

Comment:
Don't touch `Nbr` and `NbrList` as you already have a `EncodedNbr` and `EncodeNbrList` variant.

Unify the terminology to only use `Encoded` or only use `Compacted`. I personally prefer `Compacted`.

Reply:
Done.